doc/plugins.dox

```diff
@@ -186,59 +195,143 @@ Let's start with the very basic headers and add more later if necessary.
+    mandatory.
+  2. Filling GeanyPlugin::funcs with function pointers
+    - @ref GeanyPluginFuncs::init : an initialization function
+    - @ref GeanyPluginFuncs::cleanup : a finalization function
+    - @ref GeanyPluginFuncs::configure : a function that provides configuration (optional)
+    - @ref GeanyPluginFuncs::help : a function that provides help (optional)
+    - @ref GeanyPluginFuncs::callbacks : a pointer to an array of PluginCallback (optional).
+    .
+    @a init and @a cleanup are mandatory, the other ones depend on how advanced your plugin is.
+    Furthermore, the @a init and @a cleanup
+    pair is called when the user activates your plugin in the Plugin Manager. So use these to
+    do advanced initialization and teardown as to not waste resources when the plugin is not even
+    enabled.
+  3. Registering for real by calling @a geany_plugin_register(). You must pass the GeanyPlugin
+    pointer that your received and filled out as above. geany_plugin_register() also takes
+    various version numbers which are detailed below. Please <b>use the macro @a
+    GEANY_PLUGIN_REGISTER()</b> where you only specify the minimum API version. Please also
+    <b>observer the return value</b> of geany_plugin_register(). Geany may refuse to load your
+    plugin due to incompatibilities, you should then abort any extra setup.
+  4. Optionally setting user data that is passed to GeanyPlugin::funcs with geany_plugin_set_data().
+    Here you can set a data pointer that is passed back to your functions called by Geany. This
+    allows to avoid global variables which may be useful. It also allows to set instance pointer
+    to objects in case your plugin isn't written in pure C, enabling you to use member functions
+    as plugin functions.
+    You may also call this function later on, for example in your @ref GeanyPluginFuncs::init
+    routine to defer costly allocations to when the plugin is actually activated by the user.
+
+
+@subsection versions On API and ABI Versions
+As previously mentioned @a geany_plugin_register() takes a number of versions as arguments:
+  1. api_version
+  2. min_api_version
+  3. abi_version
+
+These refer to Geany's versioning scheme to manage plugin compatibility. It follows conrete rules:
+  - Plugins are compiled against a specific Geany version on the build machine at the time of plugin
+    compilation. This version of Geany has specific ABI and API versions, which will be compiled
+    into the plugin.
+  - The Geany version that loads the plugin may be different, possibly even have different API and
+    ABI versions.
+  - The ABI version is primary for plugin compatibility. The ABI version of the running Geany and
+    the one that's compiled into the plugin must match exactly (==). In case of mismatch, the
+    affected plugins generally need to be simply recompiled (without source code changes) against
+    the running Geany.
```

Comment:
That's not generally totally true.  While we do try to keep compatibility as much as possible even when we need to break ABI, it often means some API changed in an incompatible manner.  As we only have 2 version numbers, we don't differentiate "broke API in an incompatible manner" from "broke ABI".

Maybe reword "(without any, or only minimal source code changes)"